test/Test/Network/Datadog/StatsD.hs

```diff
@@ -52,3 +55,39 @@ spec = describe "StatsD spec" $ do
       case val of
         Just _  -> pure ()
         Nothing -> expectationFailure "Did not receive DogStatsD event"
+  it "does not go into an infinite loop when trying sending a metric larger than a UPD packet" $ do
+    let longText = Text.replicate 65507 "x"
```

Comment:
I think it would be better to get the `65507` value from the settings instead when you create it on line 61 with `dogStatsSettingsBufferSize`.

Reply:
I was a bit torn on that since the value is really related to the max size of a UDP package, which isn't going to change. I'll think through it and revise it somehow before I post the PR upstream.

Reply:
That's a good point. Something else you might want to consider is this line from the [datadog documentation](https://www.stackage.org/haddock/lts-12.21/datadog-0.2.3.0/Network-StatsD-Datadog.html#v:dogStatsSettingsBufferSize):
> In some scenarios, however, you may wish to send smaller packets.

Although, since you are using the default configuration, that might not be a factor for this unit test.

Reply:
Ultimately I decided to do nothing here, because the matching the actual *rendered* chunk length in statsd format to the configured buffered size is delicate, and could easily be broken in the future by changes to `renderEvent`.

Leaving the value at the maximum possible UDP packet data size guarantees that this test will produce a "too large" packet no matter what (realistic) default value is configured in the library in the future.